examples/molecular_dynamics/example_molecular_dynamics.cpp

```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Random.hpp>
+
+template <class MemorySpace>
+struct Neighbors
+{
+  Kokkos::View<ArborX::Point *, MemorySpace> particles_;
+  float cutoff_radius_;
+};
+
+template <class MemorySpace>
+struct ArborX::AccessTraits<Neighbors<MemorySpace>, ArborX::PredicatesTag>
+{
+  using memory_space = MemorySpace;
+  using size_type = std::size_t;
+  static KOKKOS_FUNCTION size_type size(Neighbors<MemorySpace> const &x)
+  {
+    return x.particles_.extent(0);
+  }
+  static KOKKOS_FUNCTION auto get(Neighbors<MemorySpace> const &x, size_type i)
+  {
+    return attach(intersects(Sphere{x.particles_(i), x.cutoff_radius_}),
+                  (int)i);
+  }
+};
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = ExecutionSpace::memory_space;
+
+  ExecutionSpace execution_space{};
+
+  float const dx = 1.f;
+  float const dy = 1.f;
+  float const dz = 1.f;
+  int const nx = 100;
+  int const ny = 100;
+  int const nz = 100;
+  int const n = 4 * nx * ny * nz;
+
+  float const r = 1.f; // cut-off radius
+
+  Kokkos::Profiling::pushRegion("setup");
+  Kokkos::View<ArborX::Point *, MemorySpace> particles(
+      Kokkos::view_alloc("points", Kokkos::WithoutInitializing), n);
+  Kokkos::parallel_for(
+      "make_particles",
+      Kokkos::MDRangePolicy<Kokkos::Rank<3>, ExecutionSpace>(
+          execution_space, {0, 0, 0}, {nx, ny, nz}),
+      KOKKOS_LAMBDA(int i, int j, int k) {
+        int const id = i * ny * nz + j * nz + k;
+        particles[4 * id + 0] = {i * dx + .0f, j * dy + .0f, k * dz + .0f};
+        particles[4 * id + 1] = {i * dx + .0f, j * dy + .5f, k * dz + .5f};
+        particles[4 * id + 2] = {i * dx + .5f, j * dy + .0f, k * dz + .5f};
+        particles[4 * id + 3] = {i * dx + .5f, j * dy + .5f, k * dz + .0f};
```

Comment:
So this will result in a slightly non-intuitive situation where you won't have particles on the right, far and top boundaries. Meaning, overall this is not a cartesian grid. Would prefer it to make the grid symmetric.

Reply:
That shouldn't matter. It is more common for this to be a periodic system, but regardless of where you stop the atoms on the faces have fewer neighbors

Reply:
Won't fix as it would add unnecessary complexity

Reply:
OK. Out of curiosity, why do the particles are only only in the middle of faces? It's equivalent to having a full 2n^3 grid with n^3 grid cutout. Does it correspond to a certain molecular structure?

Reply:
Yes, it's the face centered cubic (FCC) structure. It's among the more common structures for metals (Al, Ni, etc.)